apps/src/code-studio/components/ShareDialog.jsx

```diff
@@ -203,6 +229,24 @@ var ShareDialog = React.createClass({
                 <i className="fa fa-mobile-phone" style={{fontSize: 36}}></i>
                 <span>Send to phone</span>
               </a>
+              {isOwner && !isPublished &&
+                <button
+                  id="sharing-publish"
+                  style={styles.button}
+                  onClick={this.publish}
+                >
+                  Publish
```

Comment:
Done. The existing i18n in this file appears to be passed in from `window.dashboard.i18n`, which seems like the old way of doing things. I've gone with adding new strings via the apps locale files instead. It seems to be about 50/50 whether localized strings are pulled in from apps or dashboard under `apps/src/code-studio/`.